benchmark/infrastructure/BenchmarkToString.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel February of 2023, schlegea@informatik.uni-freiburg.de)
+
+#include "../benchmark/infrastructure/BenchmarkToString.h"
+
+#include <absl/strings/str_cat.h>
+#include <absl/strings/str_replace.h>
+
+#include <sstream>
+
+#include "BenchmarkMeasurementContainer.h"
+#include "BenchmarkMetadata.h"
+#include "util/Exception.h"
+#include "util/Forward.h"
+
+namespace ad_benchmark {
+
+/*
+ * @brief Return a string of the form
+ * "#################
+ * # categoryTitle #
+ * #################"
+ */
+static std::string createCategoryTitle(std::string_view categoryTitle) {
+  // The bar above and below the title.
+  const size_t barLength = categoryTitle.size() + 4;
+  const std::string bar(barLength, '#');
+
+  return absl::StrCat(bar, "\n# ", categoryTitle, " #\n", bar);
+}
+
+// ___________________________________________________________________________
+std::string addIndentation(const std::string_view str,
+                           const size_t& indentationLevel) {
+  // An indention level of 0 makes no sense. Must be an error.
```

Comment:
You often write `indention`, but it is `indentation`.

Reply:
Fixed in [574b0bd](https://github.com/ad-freiburg/qlever/pull/1010/commits/574b0bd0bb8bedcd7a76e2ed7c8f9bce03a2cbef).